core/arch/mangle_shared.c

```diff
@@ -309,20 +330,33 @@ cleanup_after_clean_call(dcontext_t *dcontext, clean_call_info_t *cci, instrlist
         cci = &default_clean_call_info;
         /* saved error code is currently on the top of the stack */
 
-#if (defined(X86) && defined(X64)) || defined(MACOS)
+#if defined(X86) && (defined(X64) || defined(UNIX))
     /* PR 218790: remove the padding we added for 16-byte rsp alignment */
     if (cci->should_align) {
+        int align = get_ABI_stack_alignment();
```

Comment:
Repetition of code w.r.t the calculation of emulate_dstack_offs.

Reply:
Hopefully you mean the code below this.  It is harder to move this line out b/c the local var is used in the condition too.  There is some repetition but with the emulate_dstack_offs shared I think it is pretty good now.

Reply:
Yes, code concerning emulate_dstack_offs.